### ISSUE TITLE
ZenoTOF not in CV: Invalid cvParam accession "1003293"

### ISSUE THREAD

**Issue by sneumann** [sneumann/mzR#278]
Hi, @RemyDeB reported https://github.com/sneumann/xcms/issues/670
and indeed the ZenoTOF is in 
https://github.com/HUPO-PSI/psi-ms-CV/blob/72ed99cae68f900a11cefb0d4aa7c39136484df3/psi-ms.obo#L21686
but not in the `cv.*` files in https://github.com/sneumann/mzR/tree/master/src/pwiz/data/common
This needs new files, possibly copied over from 
https://github.com/ProteoWizard/pwiz/tree/master/pwiz/data/common
which is known to have the 1003293
https://github.com/ProteoWizard/pwiz/blob/f7165e119169b38ac3af6f5d29276f88ee1e30a9/pwiz/data/common/psi-ms.obo#L21686
Yours, Steffen

**Comment by RemyDeB** [comment]
Hi, 
Do i have to create a pull request with right `cv.*` files analogous to [#210 ](https://github.com/sneumann/mzR/pull/210) ? 

Regards, Remy

**Comment by sneumann** [comment]
Hi, theoretically that would be great, but I was faster this time :-) 
https://github.com/sneumann/mzR/pull/279
BUT with that I get errors on `R CMD check mzR` because of:
```
Error: processing vignette 'mzR.Rmd' failed with diagnostics:
Can not open file /usr/local/lib/R/site-library/msdata/threonine/threonine_i2_e35_pH_tree.mzXML! 
Original error was: 
Error in eval(expr, envir, enclos): 
[CVTranslator::insert()] Collision: 1001339 1002707
```
I don't yet know what that means (I guess either the CV maintainers or some converters messed up somewhere, and now pwiz finds and barfs on that). I don't know if I need to fix the ancient test file in `msdata` package, or something in the code. 
Good news: you can install the branch https://github.com/sneumann/mzR/tree/fix/issue278 locally and fix your problem. 
Yours, 
Steffen

**Comment by sneumann** [comment]
These indeed have a very similar name:
```
[Term]
id: MS:1001339
name: (?<=[KR])
def: "Regular expression for Trypsin/P." [PSI:PI]
is_a: MS:1001180 ! Cleavage agent regular expression
```
and 
```
[Term]
id: MS:1002707
name: (?=[KR])
def: "Regular expression for LysargiNase." [PSI:PI]
is_a: MS:1001180 ! Cleavage agent regular expression
```

**Comment by RemyDeB** [comment]
Hi, Can install the branch locally but got the same error as yours when i run `readMSData`.

traceback : 
`7: stop(structure(list(message = "[CVTranslator::insert()] Collision: 1001339 1002707", 
       call = NULL, cppstack = NULL), class = c("std::runtime_error", 
   "C++Error", "error", "condition")))
6: .External(structure(list(name = "CppMethod__invoke_notvoid", 
       address = <pointer: 0x0000021596693c10>, dll = structure(list(
           name = "Rcpp", path = "C:/Users/remyd/AppData/Local/R/win-library/4.2/Rcpp/libs/x64/Rcpp.dll", 
           dynamicLookup = TRUE, handle = <pointer: 0x00007fff882a0000>, 
           info = <pointer: 0x0000021597699d00>), class = "DLLInfo"), 
       numParameters = -1L), class = c("ExternalRoutine", "NativeSymbolInfo"
   )), <pointer: 0x000002159b87c700>, <pointer: 0x0000021597fe3fc0>, 
       .pointer)
5: object@backend$getInstrumentInfo()
4: instrumentInfo(msdata)
3: instrumentInfo(msdata)
2: readOnDiskMSData(files = files, pdata = pdata, msLevel. = msLevel., 
       verbose = verbose, centroided. = centroided., smoothed. = smoothed.)
1: readMSData(dda_file, mode = "onDisk")`

Your guess seems to be right, looks like a typing error 
I don't know if it's usefull to keep cleavage agents in our case maybe can delete this lines and wait for a patch ?  

Regards, Remy

**Comment by sneumann** [comment]
Nope, I am not gonna touch (and maintain those changes throughout updates ...) the psi-ms.obo file. 
Instead, I pulled in newer CVTranslator files which seem to circumvent the issue https://github.com/sneumann/mzR/commits/fix/issue278
So for me R CMD check now passes, and if you confirm I can merge. 
Yours, Steffen

**Comment by RemyDeB** [comment]
New version works well can read `.mzXL` files of ZenoTOF.

Yours , Remy
